src/swap/SwapScreen.tsx

```diff
@@ -182,19 +183,26 @@ export function SwapScreenSection({ showDrawerTopNav }: { showDrawerTopNav: bool
         }),
       })
 
-      const fromFiatValue = swapFromAmount.multipliedBy(fromToken?.usdPrice || 0)
-      const toFiatValue = swapToAmount.multipliedBy(toToken?.usdPrice || 0)
-      const priceImpact = fromFiatValue.minus(toFiatValue).dividedBy(fromFiatValue)
-      const priceImpactExceedsThreshold = priceImpact.gte(priceImpactWarningThreshold)
+      // The estimated price impact value is a percentage, with value from 0 to 100
+      const priceImpact = exchangeRate.estimatedPriceImpact
+        ? new BigNumber(exchangeRate.estimatedPriceImpact).dividedBy(100)
+        : undefined
+
+      const isPriceImpactMissing = priceImpact === undefined
+
+      const priceImpactExceedsThreshold =
+        !!priceImpact && priceImpact.gte(priceImpactWarningThreshold)
+
+      setShowMissingPriceImpactWarning(isPriceImpactMissing)
```

Comment:
actually now that i say this, we will have 3 scenarios for price warnings in the future (if we plan to remove the fiat pricing requirement):

1. high price impact
2. null price impact
3. null fiat prices for either token

i wonder if 2 and 3 can be grouped, because there's not much user action that can be taken except to proceed with caution. perhaps we can design the copy / translation string keys for 2 to be generic and include the future scenario 3?

Reply:
Do we want to warn the user in `3`? If token prices aren't needed to swap, should the user care about them?
Maybe we want to use that data (e.g. missing token prices) as a argument to say that token is weird, but in reality, the missing token price could be an error our side (for example, the curve tokens that we are missing).

Reply:
yeah good question...i think that we should design assuming that our services work as expected, and we do have some work to support curve prices lined up. i think we also eventually want to remove the requirement that only tokens that we have prices for should be swappable since this is not what most wallets / dex's do, so this warning would be necessary. we can discuss this at our weekly this week - i just wanted to call it out here because i think it's worth thinking about a generically worded warning here that can be reused. having no price impact doesn't mean much for the user and we just want to provide an action point (which is to tread carefully, but this can be applied to other scenarios)